cluster_utils/settings.py

```diff
@@ -222,6 +248,16 @@ def sanitize_numpy_torch(possibly_np_or_tensor):
 
 
 def save_metrics_params(metrics, params):
```

Comment:
Should we rename this function to `save_metrics_and_params` (deprecating the old one)?

Also, I don't like the design of this function, as it does two different things: saves parameters and metrics (as the function name indicates), but also reports metrics to the server.

Reply:
IMO it's okay to do both in one function here, so that the user only has to add one call to their script.  But we could consider splitting it into separate internal functions, i.e. like this (with better names, of course)):
```python
def save_metrics_params(m, p):
     _save(m, p)
     _sent(m)
```

Regarding the name: For the user API I'd rather use a name that focuses less on the technical details but more on the intend.  Something like `finalize_job()`.  Likewise `read_params_from_cmdline` could be renamed to s.th. like `initialize_job()`.  I think from a user point of view, those names would make it more obvious when/why to call them.

Reply:
Yep, I like the idea of renaming the functions. Then the dual purpose in `save_metrics_params` is also fine with me. Can go to a separate PR!